Python/bltinmodule.c

```diff
@@ -1411,6 +1439,7 @@ map_traverse(mapobject *lz, visitproc visit, void *arg)
 static PyObject *
 map_next(mapobject *lz)
 {
+    Py_ssize_t i;
```

Comment:
Can you just add a comment to know that *i* is declared here for error reporting afterwards?

Reply:
```suggestion
    Py_ssize_t i; // Needed for error reporting
```

Reply:
I think you forgot to commit this.

Reply:
If you declare an iteration index variable in the beginning of a scope, it is implicit that it is needed for the entire scope of the function, IMO. Comments should IMO be reserved for the parts of the source code where it is not immediately obvious what is happening or why something was added. An index variable is IMO too trivial to deserve a comment :)

Reply:
My rationale was that ANSI C requires to declare variables at the top of the scope. Here it's because we need to remember when we stoped not because of that convention (which is used a lot by old code).